tests/sqlite/sqlite.rs

```diff
@@ -725,3 +729,83 @@ async fn concurrent_read_and_write() {
     read.await;
     write.await;
 }
+
+#[sqlx_macros::test]
+async fn test_query_with_progress_handler() -> anyhow::Result<()> {
+    let mut conn = new::<Sqlite>().await?;
+
+    // Using this string as a canary to ensure the callback doesn't get called with the wrong data pointer.
+    let state = format!("test");
+    conn.lock_handle().await?.set_progress_handler(1, move || {
+        assert_eq!(state, "test");
+        false
+    });
+
+    match sqlx::query("SELECT 'hello' AS title")
+        .fetch_all(&mut conn)
+        .await
+    {
+        Err(sqlx::Error::Database(err)) => assert_eq!(err.message(), String::from("interrupted")),
+        _ => panic!("expected an interrupt"),
+    }
+
+    Ok(())
+}
+
+#[sqlx_macros::test]
+async fn test_multiple_set_progress_handler_calls_drop_old_handler() -> anyhow::Result<()> {
+    static OBJECTS_DROPPED: AtomicUsize = AtomicUsize::new(0);
+
+    struct Handler(pub &'static str);
+    impl FnOnce<()> for Handler {
+        type Output = bool;
+
+        extern "rust-call" fn call_once(mut self, args: ()) -> bool {
+            self.call_mut(args)
+        }
+    }
+    impl FnMut<()> for Handler {
+        extern "rust-call" fn call_mut(&mut self, _args: ()) -> bool {
+            assert_eq!(3, self.0.len());
+            false
+        }
+    }
+    impl Drop for Handler {
+        fn drop(&mut self) {
+            OBJECTS_DROPPED.fetch_add(1, Ordering::Relaxed);
+        }
+    }
```

Comment:
This is unnecessary: we don't want to require nightly just to run this test. 

Just have a simple type that counts the number of times it was dropped and capture that in a normal closure, please.

Reply:
I'm not exactly sure how to create a type that implements the callback function type without these features. When I try to simply capture the state via a reference counted type, the `FnMut` requires that the captured state be `Copy` while `Drop` forbids `Copy`. Would you happen to have something concrete in mind? Thanks. 
We can drop the test as a last measure.

Reply:
@nbaztec you need to mark the closure with `move` as shown in my suggestion here: https://github.com/launchbadge/sqlx/pull/2256#discussion_r1138036923